### ISSUE TITLE
erro na chamada do método orderManager.retrieveOrders

### ISSUE THREAD

**Issue by psborges84** [DeveloperCielo/LIO-SDK-Sample-Integracao-Local#23]
estou tentando chamar esse método

ResultOrders orders = orderManager.retrieveOrders(200, 0);

só que acontece a seguinte exception

attempt to invoke virtual method 'java.util.List cielo.orders.domain.Order.getPayments()' on a null object reference

o objeto orderManager está preenchido.

**Comment by psborges84** [comment]
este método retrieveOrders não esta listando todas as ordens da lio v2. tenho uma ordem que já foi feita, inclusive recebido o pagamento, mas ela não esta sendo listada conforme a chamada do método 

ResultOrders orders = orderManager.retrieveOrders(200, 0);

**Comment by psborges84** [comment]
as ordens que eu crio no meu aplicativo na lista, mas eu consigo fazer o recebimento normalmente.

**Comment by haroldolivieri** [comment]
@psborges84 sabe me dizer se esse erro já estava ocorrendo na versão anterior? Já estamos analisando.

**Comment by haroldolivieri** [comment]
@psborges84 você garantiu que utilizou o método de listagem de ordens após ter recebido sucesso no método de `bind()` do OrderManager?

Como está descrito nessa [parte da documentação](https://developercielo.github.io/manual/cielo-lio#vincular-o-contexto-da-aplicação-ao-sdk)

**Comment by psborges84** [comment]
@haroldolivieri sim.

Fiz desta forma.
 
protected void configSDK() {
 orderManager = new OrderManager(credentials, this);
        orderManager.bind(this, new ServiceBindListener() {
            @Override
            public void onServiceBoundError(Throwable throwable) {
                orderManagerServiceBinded = false;
                Toast.makeText(getApplicationContext(),
                        String.format("Erro fazendo bind do serviço de ordem -> %s",
                                throwable.getMessage()), Toast.LENGTH_LONG).show();
            }

            @Override
            public void onServiceBound() {
                orderManagerServiceBinded = true;
            }

            @Override
            public void onServiceUnbound() {
                orderManagerServiceBinded = false;
            }
        });

private void inicializarRecebimentoCielo() {

  if (!orderManagerServiceBinded) {
            Toast.makeText(this, "Serviço de ordem ainda não recebeu retorno do método bind().\n"
                    + "Verifique sua internet e tente novamente", Toast.LENGTH_LONG).show();
            return;
        }

           final String referenciaPedido = construirIdOrdemPagamento();
            orderPagamento = orderManager.createDraftOrder(referenciaPedido);
            orderPagamento.addItem(String.valueOf(1 + (Math.random())), "TESTE", valorRecebCielo, 1, "UNITARIO");
            orderManager.updateOrder(orderPagamento);
            orderManager.placeOrder(orderPagamento);

orderManager.checkoutOrder() .... -> estou recebendo passando o parametro de email. a chamada funciona corretamente com esta nova versão 

ele cria a minha ordem corretamente, tanto e que vou no menu da LIO  Atalhos -> Vender -> Pedidos -> Consigo visualizar o meu pedido só que o status dele e em "Pagamento". Se eu vou no sdk de teste e listar as ordens ... ele não mostra essa que acabei de criar.

através do método 

ResultOrders resultOrders = orderManager.retrieveOrders(200, 0);

}

**Comment by psborges84** [comment]
Na LIO versão 1 com o sdk 0.15.0

o código abaixo funcionava corretamente..


```
   private void cancelarRecebimento() {

        try {

            if (!orderManagerServiceBinded) {
                Toast.makeText(this, "Serviço de ordem ainda não recebeu retorno do método bind().\n"
                        + "Verifique sua internet e tente novamente", Toast.LENGTH_LONG).show();
                return;
            }

            exibirDialogProgresso();

            Uri recebimentoUri = getUriRecebimentoSelecionado();
            Cursor recebimentoCursor = getContentResolver().query(recebimentoUri, null, null, null, null);
            recebimentoCursor.moveToFirst();

            int colReferencia = recebimentoCursor.getColumnIndex(RecebimentoContract.Entry.COLUMN_REFERENCIA);
            final String referencia = recebimentoCursor.getString(colReferencia); // ID DO PEDIDO NA LIO

            recebimentoCursor.close();

            StringBuilder sb = new StringBuilder();
            sb.append("cancelarRecebimento => referencia = " + referencia);
            sb.append("\n");
            sb.append("orderManager => " + orderManager);
            sb.append("\n");

            final AlertDialog.Builder alert = new AlertDialog.Builder(ReceberPagamentoCartaoActivity.this);
            alert.setTitle("Debug");
            alert.setMessage(sb.toString());
            alert.setPositiveButton("Fechar",
                    new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog,
                                            int which) {
                            dialog.dismiss();
                        }
                    });
            alert.show();

            ResultOrders resultOrders = orderManager.retrieveOrders(200, 0);
            if (resultOrders != null) {
                final List<Order> orderList = resultOrders.getResults();
                sb = new StringBuilder();
                sb.append("resultOrders = " + orderList.size());
                sb.append("\n");
                for (Order o : orderList) {
                    if (o != null) {
                        boolean searchOk = referencia.equalsIgnoreCase(o.getReference());
                        if (searchOk) {
                            orderCancelamento = o;
                            break;
                        }
                    }
                    sb.append("o.getReference() = " + o.getReference());
                    sb.append("\n");
                }
                sb.append("orderCancelamento = " + orderCancelamento);
                sb.append("\n");
                sb.append("orderCancelamento.getPayments().size() = " + orderCancelamento.getPayments().size());
                sb.append("\n");
                final AlertDialog.Builder alert1 = new AlertDialog.Builder(ReceberPagamentoCartaoActivity.this);
                alert1.setTitle("Debug");
                alert1.setMessage(sb.toString());
                alert1.setPositiveButton("Fechar",
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog,
                                                int which) {
                                dialog.dismiss();
                            }
                        });
                alert1.show();
            } else {
                Toast.makeText(this, "resultOrders = " + resultOrders.getResults().size(), Toast.LENGTH_LONG).show();
            }

            resetStateCieloCancelamento();
            fecharDialogProgresso();

        } catch (Exception erro) {
            fecharDialogProgresso();
            Toast.makeText(ReceberPagamentoCartaoActivity.this, erro.getMessage(), Toast.LENGTH_LONG).show();
        }
    }
```

**Comment by haroldolivieri** [comment]
@psborges84 você está utilizando o mesmo bind() tanto para o checkout quanto para a listagem de ordens, correto? A listagem de ordens e o checkout devem utilizar o bind assinado com as mesmas credenciais. Você me confirma isso?

**Comment by haroldolivieri** [comment]
@psborges84 você poderia testar a listagem de ordens da sua LIO utilizando a aplicação sample? Coloque as suas credenciais na classe `BasePaymentActivity` e na `ListOrdersActivity` -> Faça uma transação de qualquer tipo pelo sample -> Vá em Listagem de Ordens e veja se a transação está listada.

**Comment by psborges84** [comment]
@haroldolivieri sim, estou utilizando o mesmo bind tanto para checkout como para listagem de ordens.

as minhas credenciais estão no método configSDK() .... eu chamo este método no onCreate da minha Actitivity. eu utilizo o mesmo orderManager. Tanto é que no debug do método cancelarRecebimento() o objeto orderManager não está nulo.

**Comment by psborges84** [comment]
@haroldolivieri na minha aplicação de sample. as credenciais são as mesmas do meu projeto de produção. ja testei. toda ordem que crio no meu projeto .... ela não lista no projeto sample. mesmo eu utilizando as mesmas credenciais.

**Comment by psborges84** [comment]
como as credenciais dos dois projetos são as mesmas, a listagem de ordens do projeto sample não teria que listar as ordens que são feitas pela minha aplicação? As minhas ordens criadas eu consigo visualizar elas em pedidos na lio.

**Comment by haroldolivieri** [comment]
@psborges84 exatamente, utilizando as mesmas credenciais deve ter o mesmo comportamento. Você pode fazer o teste com o sample? Fiz aqui com as minhas credenciais e, aparentemente, funciona normalmente.

**Comment by psborges84** [comment]
erro fazendo bind do serviço de ordem java.net.SocketTimeoutException: failed to connect to sensedia API

**Comment by haroldolivieri** [comment]
@SthefaniaMFerri pode validar o problema de timeout?

**Comment by psborges84** [comment]
@haroldolivieri mas o problema não é esse. gostaria de saber porque quando eu faço uma ordem na minha aplicação eu não consigo listar ela no aplicativo sample.

**Comment by psborges84** [comment]
@haroldolivieri sendo que estão com as mesmas credenciais, esta função estava ok na versão 1 da lio. até o sdk 0.15.

**Comment by psborges84** [comment]
@haroldolivieri acabei de testar o aplicativo sample, com as minhas credencias de produção. fiz um recebimento via pagamento informando email. foi feito o pagamento. corretamente saiu o comprovante. eu fui em listagem de ordens não aparece a ordem que eu criei.. pode estar ai o problema.

**Comment by psborges84** [comment]
@haroldolivieri fui em pedido na LIO consta a ordem "Teste - Email" com o status PAGO. Mas não aparece ela na listagem de ordens

**Comment by haroldolivieri** [comment]
@psborges84 estou tentando reproduzir o problema para poder te dar uma solução. Assim que tiver algo já mando por aqui.

**Comment by psborges84** [comment]
@haroldolivieri ok no aguardo.

**Comment by psborges84** [comment]
@haroldolivieri identificou algo?

**Comment by psborges84** [comment]
@haroldolivieri ?

**Comment by haroldolivieri** [comment]
@psborges84 assim que tivermos o parecer te deixarei a par.

**Comment by psborges84** [comment]
@haroldolivieri mais é um bug? vocês identificaram algo? preciso de pelo menos uma posição. pois minha app depende desta funcionalidade para cancelar uma venda realizada.

**Comment by psborges84** [comment]
@haroldolivieri novidade?

**Comment by psborges84** [comment]
@haroldolivieri ?

**Comment by brunonmelo** [comment]
Você não está tentando `orderManager.retrieveOrders(); ` antes do bind da aplicação está finalizado não?
Se você faz requisições ao orderManager antes do bind está completo, você tera um `kotlin.KotlinNullPointerException`.
